content-src/asrouter/templates/FirstRun/FirstRun.jsx

```diff
@@ -0,0 +1,216 @@
+    };
+  },
+
+  addFluent(document) {
+    FLUENT_FILES.forEach(file => {
+      const link = document.head.appendChild(document.createElement("link"));
+      link.href = file;
+      link.rel = "localization";
+    });
+  },
+
+  async fetchFlowParams({ fxaEndpoint, UTMTerm, dispatch, setFlowParams }) {
+    try {
+      const url = new URL(
+        `${fxaEndpoint}/metrics-flow?entrypoint=activity-stream-firstrun&form_type=email`
+      );
+      addUtmParams(url, UTMTerm);
+      const response = await fetch(url, { credentials: "omit" });
+      if (response.status === 200) {
+        const { deviceId, flowId, flowBeginTime } = await response.json();
+        setFlowParams({ deviceId, flowId, flowBeginTime });
+      } else {
+        dispatch(
+          ac.OnlyToMain({
+            type: at.TELEMETRY_UNDESIRED_EVENT,
+            data: {
+              event: "FXA_METRICS_FETCH_ERROR",
+              value: response.status,
+            },
+          })
+        );
+      }
+    } catch (error) {
+      dispatch(
+        ac.OnlyToMain({
+          type: at.TELEMETRY_UNDESIRED_EVENT,
+          data: { event: "FXA_METRICS_ERROR" },
+        })
+      );
+    }
+  },
+};
+
+export class FirstRun extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.didLoadFlowParams = false;
+
+    this.state = {
+      prevMessage: undefined,
+
+      hasInterrupt: false,
+      hasTriplets: false,
+
+      interrupt: undefined,
+      triplets: undefined,
+
+      isInterruptVisible: false,
+      isTripletsContainerVisible: false,
+      isTripletsContentVisible: false,
+
+      UTMTerm: "",
+
+      flowParams: undefined,
+    };
+
+    helpers.addFluent(this.props.document);
+  }
+
+  static getDerivedStateFromProps(props, state) {
+    const { message } = props;
+    if (message !== state.prevMessage) {
+      const {
+        hasTriplets,
+        hasInterrupt,
+        interrupt,
+        triplets,
+        UTMTerm,
+      } = helpers.selectInterruptAndTriplets(message);
+
+      return {
+        prevMessage: message,
+
+        hasInterrupt,
+        hasTriplets,
+
+        interrupt,
+        triplets,
+
+        isInterruptVisible: hasInterrupt,
+        isTripletsContainerVisible: hasTriplets,
+        isTripletsContentVisible: false,
+
+        UTMTerm,
+      };
+    }
+    return null;
+  }
+
+  fetchFlowParams() {
+    const { fxaEndpoint, dispatch } = this.props;
+    const { UTMTerm } = this.state;
+    if (fxaEndpoint && UTMTerm && !this.didLoadFlowParams) {
+      this.didLoadFlowParams = true;
+      helpers.fetchFlowParams({
+        fxaEndpoint,
+        UTMTerm,
+        dispatch,
+        setFlowParams: flowParams => this.setState({ flowParams }),
+      });
+    }
+  }
+
+  componentDidMount() {
+    this.fetchFlowParams();
+
+    // We need to remove hide-main since we should show it underneath everything that has rendered
+    this.props.document.body.classList.remove("hide-main");
```

Comment:
We only want to immediately remove "hide-main" for Trailhead as otherwise the page can be scrolled for "fxa_overlay" down to only show blue background. RTAMO interstitial happens to position so it doesn't scroll.